express-notes-api/server.js

```diff
@@ -0,0 +1,171 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/notes', async (req, res) => {
+  const notesArray = [];
+  try {
+    const contents = await readFile('./data.json', 'utf8');
+    const data = JSON.parse(contents);
+    const { notes } = data;
+    for (const key in notes) {
+      notesArray.push(notes[key]);
+    }
+    if (notes) {
+      res.json(notesArray);
+    }
+  } catch (error) {
+    res.json([]);
+  }
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const errorResponse = {};
+  try {
+    const contents = await readFile('./data.json', 'utf8');
+    const data = JSON.parse(contents);
+    const { notes } = data;
+
+    if (parseInt(req.params.id) < 0) {
+      errorResponse.error = 'id must be a positive integer';
+      throw new Error();
+    }
+    if (notes[parseInt(req.params.id)] === undefined) {
+      errorResponse.error = `cannot find note with id ${req.params.id}`;
+      throw new Error();
+    }
+
+    res.status(200).json(notes[parseInt(req.params.id)]);
+    return;
+  } catch (error) {
+    switch (errorResponse.error) {
+      case 'id must be a positive integer':
+        res.status(400).send(errorResponse);
+        break;
+      case `cannot find note with id ${req.params.id}`:
+        res.status(404).send(errorResponse);
+        break;
+      default:
+        errorResponse.error = 'an unexpected error has occured';
+        res.status(500).send(errorResponse);
+        break;
+    }
```

Comment:
Since you are putting a string in errorResponse that you expect a user to read, it will eventually be localized to whatever language the end user is using. Once you do that, you won't be able to switch on its value. Also, even now, you have to exactly recreate the string, so it shows up twice.

It would be much easier to just put some arbitrary key in `errorResponse`, such as `'not-positive'` or `'not-found'`. Then you can switch on that and put the desired test directly in your response.

Or, you could make errorResponse just be the status code (400, 404, etc.). Then switch on that code.